amaranth/vendor/_lattice_nexus.py

```diff
@@ -0,0 +1,862 @@
+                    o_HFCLKOUT=clk_i,
+                )
+            else:
+                clk_i = self.request(self.default_clk).i
+
+            if self.default_rst is not None:
+                rst_i = self.request(self.default_rst).i
+            else:
+                rst_i = Const(0)
+
+            gsr0 = Signal()
+            gsr1 = Signal()
+            # On Nexus all the D-type FFs have either an synchronous or asynchronous preset. Here
+            # we build a simple reset synchronizer from D-type FFs with a positive-level
+            # asynchronous preset which we tie low
+            m.submodules += [
+                Instance(
+                    "FD1P3BX",
+                    p_GSR="DISABLED",
+                    i_CK=clk_i,
+                    i_D=~rst_i,
+                    i_SP=Const(1),
+                    i_PD=Const(0),
+                    o_Q=gsr0,
+                ),
+                Instance(
+                    "FD1P3BX",
+                    p_GSR="DISABLED",
+                    i_CK=clk_i,
+                    i_D=gsr0,
+                    i_SP=Const(1),
+                    i_PD=Const(0),
+                    o_Q=gsr1,
+                ),
+                Instance("GSR", p_SYNCMODE="SYNC", i_CLK=clk_i, i_GSR_N=gsr1),
+            ]
+            # GSR implicitly connects to every appropriate storage element. As such, the sync
+            # domain is reset-less; domains driven by other clocks would need to have dedicated
+            # reset circuitry or otherwise meet setup/hold constraints on their own.
+            m.domains += ClockDomain("sync", reset_less=True)
+            m.d.comb += ClockSignal("sync").eq(clk_i)
+            return m
+
+    # pg. 17 of FPGA-TN-02067-1-8-sysIO-User-Guide-Nexus-Platform.pdf
+    _single_ended_io_types = [
+        "LVCMOS33", "LVTTL33", 
+        "LVCMOS25", 
+        "LVCMOS18", "LVCMOS18H",
+        "LVCMOS15", "LVCMOS15H",  
+        "LVCMOS12", "LVCMOS12H", 
+        "LVCMOS10", "LVCMOS10H", "LVCMOS10R", 
+        "SSTL15_I", "SSTL15_II",
+        "SSTL135_I", "SSTL135_II", 
+        "HSTL15_I", 
+        "HSUL12",
+    ]
+    _differential_io_types = [
+        "LVCMOS33D", "LVTTL33D",
+        "LVCMOS25D", 
+        "SSTL15D_I", "SSTL15D_II",
+        "SSTL135D_I", "SSTL135D_II", 
+        "HSTL15D_I", 
+        "HSUL12D",
+        "LVDS", "LVDSE", "SUBLVDS", "SUBLVDSEH", 
+        "SLVS", 
+        "MIPI_DPHY", 
+    ]
+
+    def should_skip_port_component(self, port, attrs, component):
+        # On Nexus a differential IO is placed by only instantiating an IO buffer primitive at
+        # the PIOA or PIOC location, which is always the non-inverting pin.
+        if (
+            attrs.get("IO_TYPE", "LVCMOS25") in self._differential_io_types
+            and component == "n"
+        ):
+            return True
+        return False
+
+    def _get_xdr_buffer(self, m, pin, *, i_invert=False, o_invert=False):
```

Comment:
Note that all of this stuff will need to be completely redone as per RFC 55 — see #1295 for an example of the changes required.